Gordon360/Services/LostAndFoundService.cs

```diff
@@ -130,15 +130,20 @@ public int CreateMissingItemReport(MissingItemReportViewModel reportDetails, str
         }
 
         /// <summary>
-        /// Create an ActionsTaken for a given missing item report.
+        /// Create an action taken for the missing item report with given id
         /// </summary>
-        /// <param name="id">The id</param>
+        /// <param name="missingItemID">The id of the missing item to add an action to</param>
         /// <param name="ActionsTaken">The actions taken object to create</param>
+        /// <param name="username">The username of the person making the request</param>
         /// <returns>actionTakenID - The ID of the action taken, generated by the database.</returns>
-        public int CreateActionTaken(int id, ActionsTakenViewModel ActionsTaken)
+        /// <exception cref="ResourceCreationException">If the requesting users account can't be found, or the action fails to save</exception>
+        /// <exception cref="ResourceNotFoundException">If no missing item report exists with the given id</exception>
+        /// <exception cref="UnauthorizedAccessException">If a non-admin user attempts to create an action on a report which doesn't
+        /// belong to them</exception>
+        public int CreateActionTaken(int missingItemID, ActionsTakenViewModel ActionsTaken, string username)
         {
-            // Store user's ID number in the database
-            var account = context.ACCOUNT.FirstOrDefault(x => x.AD_Username == ActionsTaken.username);
+            // Get requesting user's ID number
+            var account = context.ACCOUNT.FirstOrDefault(x => x.AD_Username == username);
```

Comment:
FYI, this whole `get id for user by username or throw exception if account not found` can be simplified by taking a dependency on AccountService (i.e. adding `IAccountService accountService` to the primary constructor on line 15), and then calling it's `GetAccountByUsername` method that handles the "throw if null" behavior for you:

```suggestion
            var idNum = accountService.GetAccountByUsername(username).gordon_id
```